### PR TITLE
Model builder capabilities vocabulary extensions.

### PR DESCRIPTION
**Pull request by g2mula**
Documentation for new feature
https://github.com/OData/ModelBuilder/pull/9

### CODE REVIEW

File: Odata-docs/webapi/model-builder-capabilities-vocabulary.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+</Schema>
+```
+
+### [SelectSupport](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Capabilities.V1.xml#L335)
+Support for $select and nested query options within $select
+```csharp
+_ = customerConfiguration.HasSelectSupport()
+    .IsSupported(true)  // Supports $select
+    .IsInstanceAnnotationsSupported(false)  // Supports instance annotations in $select list
+    .IsExpandable(false)  // $expand within $select is supported
```
Comment (wandeg) [inline Odata-docs/webapi/model-builder-capabilities-vocabulary.md:76]: We might end up with some folks taking this literally i.e. IsExpandable(false) actually means that '$expand within $select is supported' even though this is a dummy example

File: Odata-docs/webapi/model-builder-capabilities-vocabulary.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        <PropertyValue Property="SkipSupported" Bool="false" />
+        <PropertyValue Property="ComputeSupported" Bool="false" />
+        <PropertyValue Property="Countable" Bool="true" />
+        <PropertyValue Property="Sortable" Bool="false" />
+      </Record>
+    </Annotation>
+  </Annotations>
+</Schema>
+```
+Note that if you do not specify any property, then these will not be output eg
```
Comment (wandeg) [inline Odata-docs/webapi/model-builder-capabilities-vocabulary.md:116]: Perhaps change to 'Note that only the properties specified will be output'

File: Odata-docs/webapi/model-builder-capabilities-vocabulary.md [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+title: " Non-convention model builder with capabilities vocabulary"
+description: "capabilities vocabulary"
+author: g2mula
+ms.author: g2mula
+ms.date: 8/17/2020
+---
+# Non-convention model builder with capabilities vocabulary
+**Applies To**:[!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v7.md)][!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v6.md)]
+
+The model builder now allows you to configure the [capabilities of a service](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Capabilities.V1.md). Please follow the link for more information on capabilities vocabulary.
```
Comment (habbes) [inline Odata-docs/webapi/model-builder-capabilities-vocabulary.md:11]: I think the the "Please follow the link" text should be a link.

File: Odata-docs/webapi/model-builder-capabilities-vocabulary.md [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+
+Unless otherwise specified, examples will be based on the model below.
+```csharp
+public class Customer
+{
+  public int CustomerId { get; set; }
+  public string Name { get; set; }
+}
+
+var customerConfiguration = builder.EntitySet<Customer>("Customers");
```
Comment (habbes) [inline Odata-docs/webapi/model-builder-capabilities-vocabulary.md:24]: Maybe also show where `builder` comes from in this code snippet. like a `var builder = ....` statement.

**Comment by habbes** [comment]
@g2mula  what about InsertRestrictions, UpdateRestrictions, OperationRestrictions, NavigationRestrictions?

File: Odata-docs/webapi/model-builder-capabilities-vocabulary.md [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+    .IsSupported(true)  // Supports $select
+    .IsInstanceAnnotationsSupported(false)  // Supports instance annotations in $select list
+    .IsExpandable(false)  // $expand within $select is supported
+    .IsFilterable(true)  // $filter within $select is supported
+    .IsSearchable(false)  // $search within $select is supported
+    .IsTopSupported(true)  // $top within $select is supported
+    .IsSkipSupported(false)  // $skip within $select is supported
+    .IsComputeSupported(false)  // $compute within $select is supported
+    .IsCountable(true)  // $count within $select is supported
+    .IsSortable(false);  // $orderby within $select is supported
```
Comment (habbes) [inline Odata-docs/webapi/model-builder-capabilities-vocabulary.md:83]: How come setting them to `false` still marks them as supported?

File: Odata-docs/webapi/model-builder-capabilities-vocabulary.md [hunk 6]
```diff
@@ -0,0 +1,251 @@
+---
+title: " Non-convention model builder with capabilities vocabulary"
```
[the remark this answers is not in the document]
Comment (g2mula) [reply Odata-docs/webapi/model-builder-capabilities-vocabulary.md:2]: No :-)

File: Odata-docs/webapi/model-builder-capabilities-vocabulary.md (hunk 5 above)
Comment (g2mula) [reply Odata-docs/webapi/model-builder-capabilities-vocabulary.md:83]: Fixed.

**Comment by g2mula** [comment]
> @g2mula what about InsertRestrictions, UpdateRestrictions, OperationRestrictions, NavigationRestrictions?

This is a general guide showing common scenarios to configure a capability, there are 39(?) capabilities what criteria should we use to use to omit or permit?

**Comment by g2mula** [comment]
> Missing the 'getting started part'. Does the user need to install this as a package for example?

This is an extension of the existing ModelBuilder, it does not require it own installation...

With that said, we probably need to first publish the new changes to nuget...

File: Odata-docs/webapi/model-builder-capabilities-vocabulary.md [hunk 7]
```diff
@@ -0,0 +1,251 @@
...
+description: "capabilities vocabulary"
+author: g2mula
+ms.author: g2mula
+ms.date: 8/17/2020
+---
+# Non-convention model builder with capabilities vocabulary
+**Applies To**:[!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v7.md)][!INCLUDE[appliesto-webapi](../includes/appliesto-webapi-v6.md)]
+
+The model builder now allows you to configure the [capabilities of a service](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Capabilities.V1.md). Please follow the link for more information on capabilities vocabulary.
+
```
Comment (mumbi-o) [inline Odata-docs/webapi/model-builder-capabilities-vocabulary.md:16]: Might it be useful to direct folks to a fully working sample similar to: https://github.com/OData/ODataSamples/tree/master/WebApiCore/ODataCapabilitiesVocabularySample

Just an idea?

File: Odata-docs/webapi/model-builder-capabilities-vocabulary.md [hunk 8]
```diff
@@ -0,0 +1,277 @@
...
+  </EntityContainer>
+</Schema>
+```
+
+## [CountRestrictions](https://github.com/oasis-tcs/odata-vocabularies/blob/master/vocabularies/Org.OData.Capabilities.V1.xml#L221)
+
+Restrictions on /$count path suffix and $count=true system query option
+
+```csharp
+_ = customerConfiguration.HasCountRestrictions().IsCountable(false);
```
[the remark this answers is not in the document]
Comment (g2mula) [reply Odata-docs/webapi/model-builder-capabilities-vocabulary.md:54]: Whatever you specify will be output, if you specify IsCountable false then go ahead and configure more things, the modelbuilder will output it just as you have specified. It does not make distinctions on the meaning of properties, that is for the server app to make sense of and decide on whether to look for this property (Countable) first before continuing processing.

File: Odata-docs/webapi/model-builder-capabilities-vocabulary.md (hunk 7 above)
Comment (g2mula) [reply Odata-docs/webapi/model-builder-capabilities-vocabulary.md:16]: Updated the existing sample, created a PR https://github.com/OData/ODataSamples/pull/137, please review and merge.

**Review by mumbi-o — APPROVED** [review]

**Pull request merged**